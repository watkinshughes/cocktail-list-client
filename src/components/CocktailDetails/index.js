import React, { Component } from "react";
import { Link } from "react-router-dom";
import axios from "axios";
import styles from "./styles.module.css";

class CocktailDetails extends Component {
  state = {
    data: {},
    loading: true
  };

  fetchData = () => {
    const url = `https://cocktail-list-api.herokuapp.com/cocktails/${
      this.props.match.params.id
    }`;
    axios.get(url).then(response => {
      this.setState({
        data: response.data,
        loading: false
      });
    });
  };

  componentDidMount() {
    this.fetchData();
  }

  render() {
    return (
      <article className={styles.details}>
        <h1>{this.state.data.name}</h1>
        <div className="display-linebreak">{this.state.data.ingredients}</div>
        <p>{this.state.data.preparation}</p>
        <p>{this.state.data.garnish}</p>
        <h2>
          <em>{this.state.data.category}</em>
        </h2>
        <Link to="/">← Back</Link>
      </article>
    );
  }
}

export default CocktailDetails;
