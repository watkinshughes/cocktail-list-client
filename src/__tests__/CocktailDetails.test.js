import React from "react";
import ReactDOM from "react-dom";
import { create } from "react-test-renderer";
import CocktailDetails from "../components/CocktailDetails";

const match = {
  params: {
    id: '12345'
  }
}

test("snapsot", () => {
  const c = create(<CocktailDetails match={match}/>);
  expect(c.toJSON()).toMatchSnapshot();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<CocktailDetails match={match}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});
