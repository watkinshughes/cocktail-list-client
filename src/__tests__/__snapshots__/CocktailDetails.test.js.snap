// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapsot 1`] = `
<div
  className="loading"
>
  <span
    className="visually-hidden"
  >
    Loading...
  </span>
</div>
`;
